webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -7635,14 +7726,56 @@ var userOverride = {
                     Mid conversation CSAT
                     update if dedicated parameter is introduced
                 */
-                   if (CSAT_ENABLED && !isConvRated) {
-                       enableDropdown = true;
-                       kommunicateCommons.modifyClassList(
-                           { id: ['km-csat-trigger'] },
-                           '',
-                           'n-vis'
-                       );
-                   }
+                if (CSAT_ENABLED && !isConvRated) {
+                    enableDropdown = true;
+                    kommunicateCommons.modifyClassList(
+                        { id: ['km-csat-trigger'] },
+                        '',
+                        'n-vis'
+                    );
+                }
+                if (appOptions.restartConversationByUser) {
+                    var restartConversationBtn = document.getElementById(
+                        'km-restart-conversation'
+                    );
+                    var isIterable = true;
+                    CURRENT_GROUP_DATA.groupMembers &&
+                        CURRENT_GROUP_DATA.groupMembers.map(function (member) {
+                            if (
+                                isIterable && (member.role == 2 ||
+                                member.roleType == 1) &&
+                                    member.userId ==
+                                        CURRENT_GROUP_DATA.conversationAssignee
+                            ) {
+                                isIterable = false;
+                                CURRENT_GROUP_DATA.initialBot = member;
+                            }
+                        });
+                    if (
+                        CURRENT_GROUP_DATA &&
+                        CURRENT_GROUP_DATA.initialBot &&
+                        CURRENT_GROUP_DATA.conversationAssignee &&
+                        CURRENT_GROUP_DATA.conversationAssignee ==
+                            CURRENT_GROUP_DATA.initialBot.userId
+                    ) {
+                        kommunicateCommons.modifyClassList(
+                            {id:['km-restart-conversation']},
+                            '',
+                            'n-vis'
+                        );
+                        restartConversationBtn &&
+                            restartConversationBtn.addEventListener(
+                                'click',
+                                mckMessageService.restartConversation
+                            );
+                    }else{
+                        kommunicateCommons.modifyClassList(
+                            {id:['km-restart-conversation']},
+                            'n-vis',
+                            ''
+                        );
+                    }
+                }
```

Comment:
This code doesn't look like it is needed to be inside of the map function. Can we move it outisde after the the value `initialBot` is set?

Reply:
It is outisde of map. Ignore.